app/Actions/Diagnostics/Pipes/Checks/SmallMediumExistsCheck.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App\Actions\Diagnostics\Pipes\Checks;
+
+use App\Contracts\DiagnosticPipe;
+use App\Enum\SizeVariantType;
+use App\Image\SizeVariantDimensionHelpers;
+use App\Models\SizeVariant;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+/**
+ * Check if there are some small or medium that could be generated.
+ */
+class SmallMediumExistsCheck implements DiagnosticPipe
+{
+	public const NUM_SMALL = 'num_small';
+	public const NUM_MEDIUM = 'num_medium';
+	public const NUM_SMALL2X = 'num_small2x';
+	public const NUM_MEDIUM2X = 'num_medium2x';
+	public const MAX_NUM_SMALL = 'max_num_small';
+	public const MAX_NUM_MEDIUM = 'max_num_medium';
+	public const MAX_NUM_SMALL2X = 'max_num_small2x';
+	public const MAX_NUM_MEDIUM2X = 'max_num_medium2x';
+	public const INFO_MSG = 'Info: Found %d %s that could be generated.';
+	public const INFO_LINE = '     You can use `php artisan lychee:generate_thumbs %s %d` to generate them.';
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function handle(array &$data, \Closure $next): array
+	{
+		if (!Schema::hasTable('configs') || !Schema::hasTable('size_variants')) {
+			// @codeCoverageIgnoreStart
+			return $next($data);
+			// @codeCoverageIgnoreEnd
+		}
+
+		$svHelpers = new SizeVariantDimensionHelpers();
+
+		/** @var object{num_small:int,num_medium:int,num_small2x:int,num_medium2x:int,max_num_small:int,max_num_medium:int,max_num_small2x:int,max_num_medium2x:int} $result */
+		$result = DB::query()
+		->selectSub(
+			SizeVariant::query()
+			->selectRaw('COUNT(*)')
+			->where('type', '=', SizeVariantType::SMALL),
+			self::NUM_SMALL
+		)
+		->selectSub(
+			SizeVariant::query()
+			->selectRaw('COUNT(*)')
+			->where('type', '=', SizeVariantType::SMALL2X),
+			self::NUM_SMALL2X
+		)
+		->selectSub(
+			SizeVariant::query()
+			->selectRaw('COUNT(*)')
+			->where('type', '=', SizeVariantType::MEDIUM),
+			self::NUM_MEDIUM
+		)
+		->selectSub(
+			SizeVariant::query()
+			->selectRaw('COUNT(*)')
+			->where('type', '=', SizeVariantType::MEDIUM2X),
+			self::NUM_MEDIUM2X
+		)
+		->selectSub(
+			SizeVariant::query()
+			->selectRaw('COUNT(*)')
+			->where(fn ($q) => $q
+				->when($svHelpers->getMaxWidth(SizeVariantType::SMALL) !== 0, fn ($q1) => $q1->where('width', '>', $svHelpers->getMaxWidth(SizeVariantType::SMALL)))
+				->when($svHelpers->getMaxHeight(SizeVariantType::SMALL) !== 0, fn ($q2) => $q2->orWhere('height', '>', $svHelpers->getMaxHeight(SizeVariantType::SMALL)))
```

Comment:
Just wondering if that `orWhere` in the second line will behave as expected when `where` is missing (i.e., if width in the first line is `0`)? It won't generate a syntactically invalid SQL query or something?

(I know -- I should test; I'm just not there yet)

Same question with respect to the other three size variants below...

Reply:
I would believe that Laravel is smart enough to resolve that. :)